crates/lib/src/qvm/mod.rs

```diff
@@ -32,73 +41,100 @@ impl QvmResultData {
     }
 }
 
-#[derive(Debug, Deserialize, Clone, PartialEq)]
-#[serde(untagged)]
-pub(super) enum Response {
-    Success(Success),
-    Failure(Failure),
-}
-
-#[derive(Debug, Deserialize, Clone, PartialEq)]
-pub(super) struct Success {
-    #[serde(flatten)]
-    pub(super) registers: HashMap<String, RegisterData>,
-}
-
-#[derive(Debug, Deserialize, Clone, Eq, PartialEq)]
-pub(super) struct Failure {
-    /// The message from QVM describing what went wrong.
-    pub(super) status: String,
+/// Run a Quil program on the QVM. The given [`Parameters`] are used to parameterize the value of
+/// memory locations across shots.
+pub async fn run(
+    quil: &str,
+    shots: u16,
+    addresses: HashMap<String, AddressRequest>,
```

Comment:
I like the idea of keeping the `api` module consistent with the QVM API, and in these higher level functions, I'm abstracting away stuff if it isn't useful. Is this `addresses` parameter an example of that? pyQuil's `qc.run` method doesn't provide this option, it just assumes you want all data for all readout registers, which we can just as easily do here using `Program`. Probably going to opt for that here too, unless someone has a good reason to do it differently.